schema-wizard-core/src/main/java/org/schemawizard/core/dao/impl/OracleHistoryTableQueryFactory.java

```diff
@@ -35,6 +35,14 @@ public String getSelectMigrationsSql() {
                 MIGRATION_TABLE_NAME);
     }
 
+    @Override
+    public String getSelectMigrationsStartedFromSql() {
+        return String.format("SELECT %s, %s, %s, %s "
+                        + "FROM %s WHERE %s >= ? ORDER BY VERSION DESC",
+                ID, VERSION, DESCRIPTION, APPLIED_ON,
+                MIGRATION_TABLE_NAME, VERSION);
+    }
+
```

Comment:
what if we have the following history:
| Id   | Version |
|------|---------|
| 1    | 001       |
| 2    | 004       |
| 3    | 002       |
| 4    | 005       |
| 5    | 003       |

And I would like to roll back to migration with version=004. Your query will return migrations with id 2 and 4 according to `WHERE %s >= ?` expression.
But migrations rollback functionality have to rollback all the migrations downwards requested migrations. So, if I would like to roll back to migration with version 004, the migrations should be rollback in the following order:
| Id   | Version |
|------|---------|
| 5    | 003       |
| 4    | 005       |
| 3    | 002       |
| 2    | 004       |

Hope, the issue is understood. But anyway, we can have a call to discuss it.

Reply:
I remember you mentioned this, and I thought maybe we should avoid such case by adding additional checks in MigrationAnalizer.upgradeAnalize() so if the last version of new migration less then last applied migration, we throw exception? or I can change query to something like this
```suggestion
"SELECT ID, VERSION, DESCRIPTION,APPLIED_ON  FROM MIGRATION_TABLE_NAME WHERE ID >= (SELECT MIN(ID) FROM MIGRATION_TABLE_NAME WHERE VERSION >= ?) ORDER BY ID DESC"
```

Reply:
We shouldn't avoid such case, because imagine the following situation:
1. Developer A is working on a `featureA` branch and creates `SW1703862739907FeatureA` migration
2. Developer B is working on a `featureB` branch and creates `SW1703862999999FeatureB` migration
3. Developer B completes his work before developer A and his task is moved to `Testing` stage. After testing, the `featureB` branch was successfully merged into `dev` branch
4. After some time, developer B completed his work, moved his task to `Testing` stage. And after testing, his task was merged into `dev` branch
Finally, we have a situation where migration with version=1703862999999 was applied before a migration with version=1703862739907

Reply:
we can use the following SQL, I believe:
`SELECT ID, VERSION, DESCRIPTION,APPLIED_ON  FROM MIGRATION_TABLE_NAME WHERE ID >= (SELECT ID FROM MIGRATION_TABLE_NAME WHERE VERSION = ?) ORDER BY ID DESC`
what do you think?